pkg/flowaggregator/apiserver/apiserver.go

```diff
@@ -0,0 +1,127 @@
+)
+
+func init() {
+	systeminstall.Install(scheme)
+	metav1.AddToGroupVersion(scheme, schema.GroupVersion{Version: "v1"})
+}
+
+type flowAggregatorAPIServer struct {
+	GenericAPIServer *genericapiserver.GenericAPIServer
+}
+
+func (s *flowAggregatorAPIServer) Run(stopCh <-chan struct{}) error {
+	return s.GenericAPIServer.PrepareRun().Run(stopCh)
+}
+
+func installHandlers(s *genericapiserver.GenericAPIServer, faq querier.FlowAggregatorQuerier) {
+	s.Handler.NonGoRestfulMux.HandleFunc("/flowrecords", flowrecords.HandleFunc(faq))
+	s.Handler.NonGoRestfulMux.HandleFunc("/recordmetrics", recordmetrics.HandleFunc(faq))
+	s.Handler.NonGoRestfulMux.HandleFunc("/loglevel", loglevel.HandleFunc())
+}
+
+// New creates an APIServer for running in flow aggregator.
+func New(faq querier.FlowAggregatorQuerier, bindPort int, cipherSuites []uint16, tlsMinVersion uint16) (*flowAggregatorAPIServer, error) {
+	cfg, err := newConfig(bindPort)
+	if err != nil {
+		return nil, err
+	}
+	s, err := cfg.New(Name, genericapiserver.NewEmptyDelegate())
+	if err != nil {
+		return nil, err
+	}
+	s.SecureServingInfo.CipherSuites = cipherSuites
+	s.SecureServingInfo.MinTLSVersion = tlsMinVersion
+	installHandlers(s, faq)
+	return &flowAggregatorAPIServer{GenericAPIServer: s}, nil
+}
+
+func newConfig(bindPort int) (*genericapiserver.CompletedConfig, error) {
+	secureServing := genericoptions.NewSecureServingOptions().WithLoopback()
+	authentication := genericoptions.NewDelegatingAuthenticationOptions()
+	authorization := genericoptions.NewDelegatingAuthorizationOptions().WithAlwaysAllowPaths("/healthz", "/livez", "/readyz")
+
+	// Set the PairName but leave certificate directory blank to generate in-memory by default.
+	secureServing.ServerCert.CertDirectory = ""
+	secureServing.ServerCert.PairName = Name
+	secureServing.BindAddress = net.IPv4zero
+	secureServing.BindPort = bindPort
+
+	authentication.WithClientTimeout(authenticationTimeout)
+
+	if err := secureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1"), net.IPv6loopback}); err != nil {
+		return nil, fmt.Errorf("error creating self-signed certificates: %v", err)
+	}
+	serverConfig := genericapiserver.NewConfig(codecs)
+	if err := secureServing.ApplyTo(&serverConfig.SecureServing, &serverConfig.LoopbackClientConfig); err != nil {
+		return nil, err
+	}
+	if err := authentication.ApplyTo(&serverConfig.Authentication, serverConfig.SecureServing, nil); err != nil {
+		return nil, err
+	}
+	if err := authorization.ApplyTo(&serverConfig.Authorization); err != nil {
+		return nil, err
+	}
+	if err := os.MkdirAll(path.Dir(TokenPath), os.ModeDir); err != nil {
+		return nil, fmt.Errorf("error when creating dirs of token file: %v", err)
+	}
+	if err := ioutil.WriteFile(TokenPath, []byte(serverConfig.LoopbackClientConfig.BearerToken), 0600); err != nil {
+		return nil, fmt.Errorf("error when writing loopback access token to file: %v", err)
+	}
+
```

Comment:
In the agent's apiserver, we set `serverConfig.Version`. Should we do the same here?

Reply:
It seems `serverConfig.Version` is used to support for `antctl version`. Maybe I can add this for the flow aggregator.

Reply:
Added